src/rust/engine/src/lib.rs

```diff
@@ -782,6 +785,35 @@ pub extern "C" fn materialize_directories(
   .into()
 }
 
+#[no_mangle]
+pub extern "C" fn setup_pantsd_logger(log_file_ptr: *const raw::c_char, level: u64) -> PyResult {
+  let path_str = unsafe { CStr::from_ptr(log_file_ptr).to_string_lossy().into_owned() };
+  let path = PathBuf::from(path_str);
+  LOGGER.set_pantsd_logger(path, level).into()
+}
+
+#[no_mangle]
+pub extern "C" fn setup_stderr_logger(level: u64) {
+  LOGGER
+    .set_stderr_logger(level)
+    .expect("Error setting up STDERR logger");
+}
+
+#[no_mangle]
+pub extern "C" fn write_log(msg: *const raw::c_char, level: u64) {
+  let message_str = unsafe { CStr::from_ptr(msg).to_string_lossy().into_owned() };
+  LOGGER
+    .log_from_python(&message_str, level)
+    .expect("Error logging message");
+}
+
+#[no_mangle]
+pub extern "C" fn set_log_level(level: u64) {
```

Comment:
It's unclear from the name what level this sets; it looks like it's a cap applied to both the stderr and the pantsd logging? Why don't we just set those two levels, rather than having this separate concept?

Reply:
Rust `log` crate has a function called `set_max_level` which changes a global (hidden) filter for logging. This is done for efficiency, to discard logs that wouldn't be logged anywhere (if I understand correctly). There are places where we want to explicitly call that function to modify the global filter to something different than what it had at startup, and this function does that. I will change the names and remove the spurious `level` field in `logger.rs::Logger` to reflect that.